src/sql/src/plan/statement/ddl.rs

```diff
@@ -775,15 +775,30 @@ fn kafka_sink_builder(
         .key_writer_schema()
         .map(|key_schema| key_schema.canonical_form());
 
-    // Use the user supplied value for replication factor, or default to 1
+    // Use the user supplied value for partition count, or default to -1 (broker default)
+    let partition_count = match with_options.remove("partition_count") {
+        None => -1,
+        Some(Value::Number(n)) => n.parse::<i32>()?,
+        Some(_) => bail!("partition count for sink topics has to be an integer"),
+    };
+
+    if partition_count == 0 || partition_count < -1 {
+        bail!(
+            "partition count for sink topics has to be a positive integer or -1 for broker default"
+        );
+    }
+
+    // Use the user supplied value for replication factor, or default to -1 (broker default)
     let replication_factor = match with_options.remove("replication_factor") {
-        None => 1,
-        Some(Value::Number(n)) => n.parse::<u32>()?,
-        Some(_) => bail!("replication factor for sink topics has to be a positive integer"),
```

Comment:
oh, I guess it already was "has to be" here. Well, change it if you want, I guess :)